components/api-server/src/client.rs

```diff
@@ -167,6 +177,48 @@ impl Client {
             }
         }
 
+        let job_config = self.get_job_config(search_job_id).await?;
+        if job_config.write_to_file {
+            Ok(SearchResultStream::File {
+                inner: self.fetch_results_from_file(search_job_id),
+            })
+        } else {
+            self.fetch_results_from_mongo(search_job_id)
+                .await
+                .map(|s| SearchResultStream::Mongo { inner: s })
+        }
```

Comment:
```suggestion
        if job_config.write_to_file {
            return Ok(SearchResultStream::File {
                inner: self.fetch_results_from_file(search_job_id),
            });
        }
        self.fetch_results_from_mongo(search_job_id)
            .await
            .map(|s| SearchResultStream::Mongo { inner: s })
```
Nit: I think in Rust, this if-else is actually treated as a single return statement. However, for the purpose of reducing indentations, we may want to write it this way, lol